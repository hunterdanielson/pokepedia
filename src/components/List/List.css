@import '../../styles/variables.css';

.List-light {
  background-color: var(--light-bg-color);
  color: var(--light-fg-color);
}

.List-dark {
  background-color: var(--dark-bg-color);
  color: var(--dark-fg-color);
}