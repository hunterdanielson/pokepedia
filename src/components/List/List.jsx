import React from 'react';
import { useList, useTheme } from '../../hooks/appContext';
import styles from './List.css';

const List = () => {
  const theme = useTheme();
  const list = useList();

  const listItems = list.map(item => (
    <li key={item.url}>
      {item.name}
    </li>
  ));
  return (
    <div className={styles[`List-${theme}`]}>
      <ul>
        {listItems}
      </ul>
    </div>
  );
};

export default List;
