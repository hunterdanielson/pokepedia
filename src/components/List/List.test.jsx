import React from 'react';
import { shallow } from 'enzyme';
import List from './List';
import AppProvider from '../../providers/appProvider';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<AppProvider><List /></AppProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
