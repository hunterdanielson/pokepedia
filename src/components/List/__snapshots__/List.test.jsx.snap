// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [Function],
      "state": Object {
        "list": Array [],
        "pageNum": 1,
        "theme": "light",
        "totalCount": 964,
      },
    }
  }
>
  <button
    name="next"
    onClick={[Function]}
  >
    Next
  </button>
  <List />
</ContextProvider>
`;
